src/olympia/abuse/admin.py

```diff
@@ -251,11 +322,52 @@ def get_queryset(self, request):
                 'addon', queryset=Addon.objects.all().only_translations()),
         )
 
+    def get_fieldsets(self, request, obj=None):
+        if obj.addon:
+            target = 'addon'
+        elif obj.user:
+            target = 'user'
+        else:
+            target = 'guid'
+        dynamic_fieldset = (
+            (None, {'fields': self.dynamic_fieldset_fields[target]}),
+        )
+        return dynamic_fieldset + self.fieldsets
+
     def target_name(self, obj):
         name = obj.target.name if obj.target else obj.addon_name
         return '%s %s' % (name, obj.addon_version or '')
     target_name.short_description = ugettext('User / Add-on')
 
+    def addon_card(self, obj):
+        template = loader.get_template('reviewers/addon_details_box.html')
+        addon = obj.addon
+        try:
+            approvals_info = addon.addonapprovalscounter
+        except AddonApprovalsCounter.DoesNotExist:
+            approvals_info = None
+        developers = addon.listed_authors
+
+        context = {
+            'addon': addon,
+            'addon_name': addon.name,
+            'approvals_info': approvals_info,
+            'reports': Paginator(
+                (AbuseReport.objects
+                    .filter(Q(addon=addon) | Q(user__in=developers))
+                    .order_by('-created')), 5).page(1),
+            'user_ratings': Paginator(
+                (Rating.without_replies
+                    .filter(addon=addon, rating__lte=3, body__isnull=False)
+                    .order_by('-created')), 5).page(1),
```

Comment:
~~Just for me to understand this better maybe, why are you using `Paginator` here instead of manually limiting?~~

Nevermind, it's to be able to render `addon_details_box.html ` without any additional changes. Maybe worth a comment, not sure.

Reply:
316252d4f831a0e99bbd0e0e5f6b588764663630